test/parallel_api/iterator/transform_iterator.pass.cpp

```diff
@@ -182,9 +182,58 @@ test_copyable()
 }
 #endif // TEST_DPCPP_BACKEND_PRESENT
 
+struct noop
+{
+    int
+    operator()(int a) const
+    {
+        return a;
+    }
+};
+
+struct noop_nodefault
+{
+    noop_nodefault() = delete;
+    noop_nodefault(int) {}
+    int
+    operator()(int a) const
+    {
+        return a;
+    }
+};
+
+void
+test_default_constructible()
+{
+    auto transformation = [](int) { return 0; };
+
+    int* ptr = nullptr;
+    oneapi::dpl::transform_iterator<int*, decltype(transformation)> trans1{ptr, transformation};
+    //default constructibility of lambdas depends on c++ standard, we want transform iterator to match its template args
+    EXPECT_TRUE((std::is_default_constructible_v<decltype(transformation)> ==
+                 ::std::is_default_constructible_v<decltype(trans1)>),
+                "transform_iterator with lambda does not match default constructibility status of the lambda itself");
+
+    //both types are default constructible
+    oneapi::dpl::transform_iterator<int*, noop> trans2{ptr, noop{}};
+    EXPECT_TRUE(::std::is_default_constructible_v<decltype(trans2)>,
+                "transform_iterator with default constructible lambda is seen to be non-default constructible");
+
+    //functor is not default constructible
+    oneapi::dpl::transform_iterator<int*, noop_nodefault> trans3{ptr, noop_nodefault{1}};
+    EXPECT_TRUE(!::std::is_default_constructible_v<decltype(trans3)>,
+                "transform_iterator with non-default constructible lambda is seen to be default constructible");
+
+    oneapi::dpl::transform_iterator<decltype(trans3), noop> trans4{trans3, noop{}};
+    EXPECT_TRUE(
+        !::std::is_default_constructible_v<decltype(trans4)>,
+        "transform_iterator with non-default constructible iterator source is seen to be default constructible");
+}
+
 std::int32_t
 main()
 {
+    test_default_constructible();
```

Comment:
Do you mean add `transform_iterator_traits.pass.cpp` or something like that to run the `test_default_constructible` and `test_copyable` ?

Reply:
transform_iterator_constructible.pass.cpp
transform_iterator_copyable.pass.cpp with skip if `TEST_DPCPP_BACKEND_PRESENT` is undefined

Reply:
OK.  with the rest of transform_iterator.pass.cpp also depending on `TEST_DPCPP_BACKEND_PRESENT`, do you think its necessary to pull `test_copyable()` out in addition to `transform_iterator_constructible.pass.cpp`?

Reply:
Yes, but I mean when this test passed you may don't know have you checked `test_copyable()` or no.
Because you calls `return TestUtils::done();` in all cases.
But up to you.

Reply:
If we pull out `transform_iterator_constructible`, then I will restore the previous done() call here.

Reply:
I've pulled out `transform_iterator_constructible.pass`, now the PR doesn't touch `transform_iterator.pass`.

Reply:
ok